File: app/src/main/java/cn/ifafu/ifafu/mvp/syllabus_item/SyllabusItemModel.java
package cn.ifafu.ifafu.mvp.syllabus_item;

import android.content.Context;

import cn.ifafu.ifafu.data.entity.Course;
import cn.ifafu.ifafu.data.entity.SyllabusSetting;
import cn.ifafu.ifafu.mvp.base.BaseModel;

class SyllabusItemModel extends BaseModel implements SyllabusItemContract.Model {

    SyllabusItemModel(Context context) {
        super(context);
    }

    @Override
    public SyllabusSetting getSyllabusSetting() {
        return repository.getSyllabusSetting();
    }

    @Override
    public void save(Course course) {
        repository.saveCourse(course);
    }

    @Override
    public void delete(Course course) {
        repository.deleteCourse(course);
    }

    @Override
    public Course getCourseById(long id) {
        return repository.getCourseById(id);
    }
}

File: app/src/main/java/cn/ifafu/ifafu/mvp/syllabus_item/SyllabusItemModel.kt
package cn.ifafu.ifafu.mvp.syllabus_item

import android.content.Context

import cn.ifafu.ifafu.base.mvp.BaseModel
import cn.ifafu.ifafu.entity.Course
import cn.ifafu.ifafu.entity.SyllabusSetting

internal class SyllabusItemModel(context: Context) : BaseModel(context), SyllabusItemContract.Model {

    override fun getSyllabusSetting(): SyllabusSetting {
        return mRepository.getSyllabusSetting()
    }

    override fun save(course: Course) {
        course.account = mRepository.getInUseUser()!!.account
        mRepository.saveCourse(course)
    }

    override fun delete(course: Course) {
        mRepository.deleteCourse(course)
    }

    override fun getCourseById(id: Long): Course? {
        return mRepository.getCourseById(id)
    }
}
